mixins/localize-mixin.js

```diff
@@ -1,189 +1 @@
-			params = arguments[1];
-		} else {
-			// legacy support for localize-behavior replacements as many args
-			for (let i = 1; i < arguments.length; i += 2) {
-				params[arguments[i]] = arguments[i + 1];
-			}
-		}
-
-		const translatedMessage = new IntlMessageFormat(value, language);
-		let formattedMessage = value;
-		try {
-			if (Object.values(params).some(v => typeof v === 'function')) throw 'localize() does not support rich text.';
-			formattedMessage = translatedMessage.format(params);
-		} catch (e) {
-			console.error(e);
-		}
-
-		return formattedMessage;
-	}
-
-	localizeHTML(key, params = {}) {
-
-		const { language, value } = this.__resources?.[key] ?? {};
-		if (!value) return '';
-
-		const translatedMessage = new IntlMessageFormat(value, language);
-		let formattedMessage = value;
-		try {
-			formattedMessage = translatedMessage.format({
-				b: chunks => localizeMarkup`<strong>${chunks}</strong>`,
-				br: () => localizeMarkup`<br>`,
-				em: chunks => localizeMarkup`<em>${chunks}</em>`,
-				i: chunks => localizeMarkup`<em>${chunks}</em>`,
-				p: chunks => localizeMarkup`<p>${chunks}</p>`,
-				strong: chunks => localizeMarkup`<strong>${chunks}</strong>`,
-				...params
-			});
-			validateMarkup(formattedMessage);
-		} catch (e) {
-			console.error(e);
-		}
-
-		return formattedMessage;
-	}
-
-	static _generatePossibleLanguages(config) {
-
-		if (config?.useBrowserLangs) return navigator.languages.map(e => e.toLowerCase()).concat('en');
-
-		const { language, fallbackLanguage } = this.documentLocaleSettings;
-		const langs = [ language, fallbackLanguage ]
-			.filter(e => e)
-			.map(e => [ e.toLowerCase(), e.split('-')[0] ])
-			.flat();
-
-		return Array.from(new Set([ ...langs, 'en-us', 'en' ]));
-	}
-
-	static _getAllLocalizeResources(config = this.localizeConfig) {
-		let resourcesLoadedPromises = [];
-		const superCtor = Object.getPrototypeOf(this);
-		// get imported terms for each config, head up the chain to get them all
-		if ('_getAllLocalizeResources' in superCtor) {
-			const superConfig = Object.prototype.hasOwnProperty.call(superCtor, 'localizeConfig') && superCtor.localizeConfig.importFunc ? superCtor.localizeConfig : config;
-			resourcesLoadedPromises = superCtor._getAllLocalizeResources(superConfig);
-		}
-		if (Object.prototype.hasOwnProperty.call(this, 'getLocalizeResources') || Object.prototype.hasOwnProperty.call(this, 'resources')) {
-			const possibleLanguages = this._generatePossibleLanguages(config);
-			const res = this.getLocalizeResources(possibleLanguages, config);
-			resourcesLoadedPromises.push(res);
-		}
-		return resourcesLoadedPromises;
-	}
-
-	_hasResources() {
-		return this.constructor['getLocalizeResources'] !== undefined;
-	}
-
-	_onResourcesChange() {
-		/** @ignore */
-		this.dispatchEvent(new CustomEvent('d2l-localize-resources-change'));
-	}
-
-});
+export { _LocalizeMixinBase as LocalizeMixin } from './localize/localize-mixin.js';
```

Comment:
Ideally [all the usages of this](https://github.com/search?type=code&q=saved%3A%22D2L+Code%22+%22import+%7B+LocalizeMixin+%7D+from+%27%40brightspace-ui%2Fcore%2Fmixins%2Flocalize-mixin.js%27%22&saved_searches=%5B%7B%22name%22%3A%22D2L+Code%22%2C%22query%22%3A%22org%3ABrightspace+OR+org%3ABrightspaceUI+OR+org%3ABrightspaceHypermediaComponents+OR+org%3ABrightspaceUILabs%22%7D%5D&expanded_query=org%3ABrightspace+org%3ABrightspaceUI+org%3ABrightspaceHypermediaComponents+org%3ABrightspaceUILabs+%22import+%7B+LocalizeMixin+%7D+from+%27%40brightspace-ui%2Fcore%2Fmixins%2Flocalize-mixin.js%27%22) would switch over to the dynamic mixin.